roles/nginx.nix

```diff
@@ -0,0 +1,37 @@
+{ config, lib, ...}: {
+  networking.firewall.allowedTCPPorts = [ 443 80 ];
+
+  services.nginx = {
+    enable = true;
+
+    recommendedGzipSettings = lib.mkDefault true;
+    recommendedOptimisation = lib.mkDefault true;
+    recommendedProxySettings = lib.mkDefault true;
+    recommendedTlsSettings = lib.mkDefault true;
+
+    # Nginx sends all the access logs to /var/log/nginx/access.log by default.
+    # instead of going to the journal!
+    commonHttpConfig = ''
+      access_log syslog:server=unix:/dev/log;
+      # Should we have this?
+      #  add_header Strict-Transport-Security 'max-age=31536000; includeSubDomains; preload' always;
```

Comment:
Only if TLS/ACME is enabled

Reply:
Only issue is that it's hard to tell when this is the case. Servers might have many vhosts and with conflicting settings.

Reply:
Leaving it out know, since enabling hsts can be difficult when you are trying to turn it off again.